lib/models/room.js

```diff
@@ -10,6 +10,19 @@ var MatrixEvent = require("./event").MatrixEvent;
 var utils = require("../utils");
 var ContentRepo = require("../content-repo");
 
+function synthesizeReceipt(userId, event, receiptType) {
+    // This is really ugly because JS has no way to express an object literal
+    // where the name of a key comes from an expression
+    var fakeReceipt = {content: {}};
+    fakeReceipt.content[event.getId()] = {};
+    fakeReceipt.content[event.getId()][receiptType] = {};
+    fakeReceipt.content[event.getId()][receiptType][userId] = {
+        ts: event
+    };
+    return new MatrixEvent(fakeReceipt);
```

Comment:
Hmm, this is contrary to your documentation, I think? https://github.com/matrix-org/matrix-js-sdk/blob/master/lib/models/room.js#L339